### ISSUE TITLE
OpenCV Models

### ISSUE THREAD

**Issue by arpit-dwivedi** [Learn-Write-Repeat/Contribution-program#24]
Add in the comments the links of resources you found and also add these things:

1. Algorithms used for each given link
2. Libraries used
3. Approx lines of codes.

In the end, also conclude which one is better.

**Comment by Rishav-hub** [comment]
### Attached file **[text_extraction.zip](https://github.com/Learn-Write-Repeat/Contribution-program/files/5358443/text_extraction.zip)**
- After downloading this file initialize your virtual environment -> `pip install pytesseract` install it if not installed,-> download TesseractOCR from [here](https://github.com/tesseract-ocr/tesseract) ->there is a .ipynb file Text_Extraction.ipynb all the code are written there -> the dataset are named as text, text1.

- [TesseractOCR](https://nanonets.com/blog/ocr-with-tesseract/) - This is a type of ***OCR*** ( Optical character recognization) which is developed by Google. Tesseract includes a new neural network subsystem configured as a text line recognizer. It basically uses the concepts of LSTM's (Long Short Term Memory) and RNN's. 
- But, before using this library there are certain preprocessing steps that are needed to be followed. Those are [Smoothing image](https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_imgproc/py_filtering/py_filtering.html), [threshold to obtain binary image](https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_imgproc/py_thresholding/py_thresholding.html?highlight=threshold), [removing noise](https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_video/py_bg_subtraction/py_bg_subtraction.html?highlight=morphologyex), [dilation](https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_imgproc/py_morphological_ops/py_morphological_ops.html?highlight=dilation), 
- Basically, the process of extraction goes like this, 
![image](https://user-images.githubusercontent.com/57321948/95500062-e2023400-09c3-11eb-8769-818ce48bc8e6.png)
- There are more steps that are included for better extraction that would be covered while writing code. They're also the flow of how LSTM works in Tesseract in this article. 
- The libraries used are ***cv2, NumPy, pytesseract***( had to be installed using pip in venv). 
- The Processing part itself requires ***20 - 27*** lines. And the contour detection and the drawing of the rectangle part requires about ***15 - 20*** lines. All about ***35 - 47*** lines.

**Comment by JulianBenny** [comment]
I started of by reading this [medium article](https://medium.com/visionwizard/object-tracking-675d7a33e687) to get a basic overview of Object tracking. This gives us a basic idea of the same and has few links to research papers and their codes in which object tracking has been implemented using different algorithms.

[**Tracking without bells and whistles - ICCV 2019**](https://arxiv.org/pdf/1903.05625.pdf)
This research paper is on the problem of tracking multiple objects in a video sequence. It uses **"tracking by detection"** algorithm to solve the same.Their tracker is called **Tracktor**. Their algorithm claims to have great results in MOT Challenge.
Their github repo is properly documented for us to refer.  

[**Tracking Objects as Points**](https://arxiv.org/pdf/2004.01177.pdf)
This paper as the name suggests tracks objects as **points**.
Their model takes the current frame, the previous frame, and a heatmap rendered from previous tracking results as input, and predicts the current detection heatmap as well as their offsets to centers in the previous frame.

[**Simple object tracking with Open CV**](https://www.pyimagesearch.com/2018/07/23/simple-object-tracking-with-opencv/)
This article uses **centroid tracking** algorithm.
-Accept bounding box coordinates and compute centroids
-Compute Euclidean distance between new bounding boxes and existing objects
-Update (x,y) coordinates of existing objects
-Register new objects
-De-register old objects.
This article guides you step by step, for the whole algorithm with clear explanations for the code as well.

This last article seems like a good choice because of all the explanations. But their algorithm seems a little basic.
This is my initial research, so making final decision on which is better will be a naive choice as of now.

**Comment by JulianBenny** [comment]
**Object Tracking**

[object-tracking.zip](https://github.com/Learn-Write-Repeat/Contribution-program/files/5358507/object-tracking.zip)

Navigate to the folder where you have downloaded the above file.
Run this command, `python object_tracker.py --prototxt deploy.prototxt \
	--model res10_300x300_ssd_iter_140000.caffemodel`
